### PR TITLE
Allow searching by care pathway type

### PR DESCRIPTION
**Pull request by jcmorrow**
I think I also need to write up a better description of how this search works, so that it's more useful to people using the platform.

### CODE REVIEW

File: src/components/text_item.tsx [hunk 1]
```diff
@@ -33,7 +33,10 @@ function highlight(props: Props, searchTerm: string) {
       key = searchKey;
       searchValue = keyValue.trim();
     }
-    if (key === searchKey && value.toLowerCase().trim().includes(searchValue)) {
+    if (
```
Comment (liangmicha) [inline src/components/text_item.tsx:36]: Do you have a sense of what lines 23 to 36 are doing?

What is the difference between `searchKey` and a `searchValue` and `value`?

File: src/util/search.ts [hunk 2]
```diff
@@ -7,11 +7,12 @@ export interface DateRange {
...
 
-const CLAIM_SEARCH_KEYS_MAP: { [key: string]: (keyof ClaimEntry)[] } = {
-  username: ["username"],
-  patient: ["patientFirstName", "patientLastName"],
+const CLAIM_SEARCH_KEYS_MAP: Record<string, (keyof ClaimEntry)[]> = {
   item: ["items"],
+  patient: ["patientFirstName", "patientLastName"],
   phone: ["phone"],
+  type: ["claimType"],
+  username: ["username"],
```
Comment (liangmicha) [inline src/util/search.ts:15]: I'm going to guess these are things that the user can search for that will match to a claim

**Review by liangmicha — APPROVED** [review]

File: src/components/text_item.tsx (hunk 1 above)
Comment (jcmorrow) [reply src/components/text_item.tsx:36]: Yeah, I eventually figured it out from playing with the code and seeing what happens. This it supposed to highlight parts of the UI that matched the search. So, if your search was "username: maisha" and we have two attendants, `maisha_owner` and `maisha_attendant` (for example) then we would highlight the `maisha` in both of those phrases when displaying them. In that case, the `searchKey` is the part of the search term before the `:` (so `username` here), the `searchValue` is the part after the `:` (so `maisha` here) and `value` is the value of the item we are currently iterating over (so for one entry it is `maisha_owner` and another time it is `maisha_attendant`). Does that make sense?

File: src/util/search.ts (hunk 2 above)
Comment (jcmorrow) [reply src/util/search.ts:15]: Yeah, that's right, and it also hooks up to `textField` to highlight the parts of the terms that are matched.

File: src/components/text_item.tsx (hunk 1 above)
Comment (liangmicha) [reply src/components/text_item.tsx:36]: gotcha, makes sense to me!

**Pull request merged**